### DISCUSSION TITLE
Requesting the Data as Consumer

### DISCUSSION THREAD

**Discussion by pandora-1988** [International-Data-Spaces-Association/DataspaceConnector#86] [Q&A] [answered]
Hi all,
I'm trying to use the IDS as Data Consumer following the official documentation.  After created the Contract Agreement for the requested artifact, I need to access the data but I have a problem with the path param "id" used in the API call GET _/api/artifacts/{id}/data/**_ at the Step 4.  If I use the UUID of the "target" URL (in green below) from the response in the previous step (Negotiate a Contract Agreement), the API returns “resource not found” because this is the **remote_id** of the required artifact.
![image](https://user-images.githubusercontent.com/73345702/128200631-f78e930b-3f99-4aec-8a51-742f0b7c9e1a.png)
The IDS does a query on the local ID and not on the remote ID. So, If I use the UUID local of the artifact, then the API returns a response with a valid body. 
It seems little bit confusing for me because I don't know how to extract this information in order to use it in the API call and I didn't find any suggestion in the documentation about that. 
Any suggestion about the last step to query the data that I want to consume?
Thanks in advance for any hint. 
@jpieperbeck @hqarawlus  @ronjaquensel

**Comment by juliapampus** [comment, marked as answer]
Hi, your observations are correct. The ID in the contract agreement is that of the provider. To get the artifact ID that the consumer needs for the GET request, the artifacts for the agreement must be viewed. In your screenshot, you see `artifacts` under `_links`. With a call on this, you get all linked artifacts with the `self` pointing to the right id. Requesting one concrete artifact next, the attribute `data` of this even includes the correct `/data` link.

**Reply by pandora-1988** [reply]
Hi Julia, thanks a lot for your suggestions, I solved the problem. Anna
